app/controllers/providers/emails_controller.rb

```diff
@@ -17,18 +17,23 @@ def update
           notice: "Email link will be to: #{citizens_legal_aid_application_url(legal_aid_application.generate_secure_id)}"
         )
       else
-        render :show
+        render :edit
       end
     end
 
     private
 
-    def edit_params
+    def update_params
       email_params.merge(model: applicant)
     end
 
     def email_params
       params.require(:applicant).permit(:email)
     end
+
+    def current_params
+      return nil unless applicant
```

Comment:
**[nit pick/not a blocker]** (something I forgot to mention in https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/100). `nil` is irrelevant here, `return unless applicant` already returns nil 💅  .

Reply:
Good point. I'll fix it in my PR